src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Globals.cs

```diff
@@ -34,6 +34,8 @@ internal static partial class Globals
         internal static Type TypeOfVoid => field ??= typeof(void);
         internal static Type TypeOfByteArray => field ??= typeof(byte[]);
         internal static Type TypeOfTimeSpan => field ??= typeof(TimeSpan);
+        internal static Type TypeOfDateOnly => field ??= typeof(DateOnly);
+        internal static Type TypeOfTimeOnly => field ??= typeof(TimeOnly);
         internal static Type TypeOfGuid => field ??= typeof(Guid);
         internal static Type TypeOfDateTimeOffset => field ??= typeof(DateTimeOffset);
         internal static Type TypeOfDateTimeOffsetAdapter => field ??= typeof(DateTimeOffsetAdapter);
```

Comment:
It's highly doubtful that caching the result of `typeof()` is helpful; we can remove it either now or in a separate PR.

Reply:
Yes, caching typeof like this is a de-optimization in current .NET. It would be best to delete these fields and use ` typeof` directly in all places they are used.